src/Workspaces/Core/Portable/EmbeddedLanguages/RegularExpressions/LanguageServices/RegexEmbeddedHighlighter.cs

```diff
@@ -0,0 +1,159 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.EmbeddedLanguages.Common;
+using Microsoft.CodeAnalysis.EmbeddedLanguages.LanguageServices;
+using Microsoft.CodeAnalysis.EmbeddedLanguages.VirtualChars;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.EmbeddedLanguages.RegularExpressions.LanguageServices
+{
+    using RegexToken = EmbeddedSyntaxToken<RegexKind>;
+
+    internal class RegexEmbeddedHighlighter : IEmbeddedHighlighter
```

Comment:
>RegexEmbeddedHighlighter [](start = 19, length = 24)

Could you add a couple tests to cover?

Reply:
I have.  They're in: https://github.com/dotnet/roslyn/pull/25517/files#diff-7dc2a5dc46eb0a831a3ad0cdf0a23f94